### PR TITLE
fix: update koishi to 4.15.3

### PR DESCRIPTION
**Pull request by MaikoTan**

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -104,13 +105,32 @@ export function defineVariation<
...
 
+  function normalizeInject(inject: string[] | Partial<Inject>): Inject {
+    if (typeof inject === 'undefined') {
+      return {
+        required: ['canvas'],
+        optional: [],
+      }
+    } else if (Array.isArray(inject)) {
+      return {
+        required: Array.from(new Set(inject)).concat(['canvas']),
```
Comment (shigma) [inline packages/core/src/index.ts:116]: 如果 inject 已经包含了 canvas 会导致重复吧？

File: packages/core/src/index.ts [hunk 2]
```diff
@@ -104,13 +105,32 @@ export function defineVariation<
...
+
   return class {
     _variation = variation
     // re-export koishi fields
     name = variation.name
     static Config = variation.Config
-    static schema = variation.schema
-    static using = Array.from(new Set(['canvas', ...(variation.using ?? [])]))
+    static inject = normalizeInject(variation.inject)
+    static using = normalizeInject(variation.using)
```
Comment (shigma) [inline packages/core/src/index.ts:133]: 已经 normalize 的情况下只需要写一个 inject 即可？

**Review by shigma — CHANGES_REQUESTED** [review]

File: packages/core/src/index.ts (hunk 1 above)
Comment (MaikoTan) [reply packages/core/src/index.ts:116]: Oh, I should wrap the `canvas` into Set as well.

File: packages/core/src/index.ts (hunk 2 above)
Comment (MaikoTan) [reply packages/core/src/index.ts:133]: Yes, you're right.

**Pull request merged**